pkg/chains/formats/intotoite6/taskrun/material.go

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright 2022 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package taskrun
+
+import (
+	"reflect"
+	"strings"
+
+	slsa "github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/v0.2"
+	"github.com/tektoncd/chains/pkg/artifacts"
+	"github.com/tektoncd/chains/pkg/chains/formats/intotoite6/attest"
+	"github.com/tektoncd/chains/pkg/chains/objects"
+	"github.com/tektoncd/pipeline/pkg/apis/resource/v1alpha1"
+	"go.uber.org/zap"
+)
+
+const (
+	uriSeparator    = "@"
+	digestSeparator = ":"
+)
+
+func isDuplicateMaterial(mats []slsa.ProvenanceMaterial, m slsa.ProvenanceMaterial) bool {
+	for _, mat := range mats {
+		if reflect.DeepEqual(m, mat) {
+			return true
+		}
+	}
+	return false
+}
+
+// add all step and sidecar images and any Git specification to materials
+func materials(tro *objects.TaskRunObject, logger *zap.SugaredLogger) []slsa.ProvenanceMaterial {
+	var mats []slsa.ProvenanceMaterial
+
+	// add step images
+	for _, stepState := range tro.Status.Steps {
+		m := imageIDToMaterial(stepState.ImageID)
+		if !isDuplicateMaterial(mats, m) {
+			mats = append(mats, m)
+		}
+	}
+
+	// add sidecar images
+	for _, sidecarState := range tro.Status.Sidecars {
+		m := imageIDToMaterial(sidecarState.ImageID)
+		if !isDuplicateMaterial(mats, m) {
+			mats = append(mats, m)
+		}
+	}
```

Comment:
Can we extract those to somewhere that can be shared with pipeline level provenance i.e. extract/extract.go?

When we do pipeline level materials, we also need to look into individual taskruns to look for the base images.

Reply:
Yes, can do!